src/internal/storage/chunk/storage.go

```diff
@@ -76,3 +77,49 @@ func (s *Storage) List(ctx context.Context, cb func(id ID) error) error {
 func (s *Storage) NewDeleter() track.Deleter {
 	return &deleter{}
 }
+
+// StableHash returns a stable hash for a set of data refs.
+func (s *Storage) StableHash(ctx context.Context, dataRefs []*DataRef) ([]byte, error) {
```

Comment:
I don't think this really belongs in the chunk package.  Can we move this to the fileset package?  It's a little weird that we have to punch through all the way to the chunk level to do this in the pfs server.

Reply:
I think this functionality makes sense in the chunk layer because it requires knowledge about how the data refs are generated by the writers and how they correspond to chunks, but I agree that we shouldn't need to punch into the chunk layer at the PFS layer. How about if I add a wrapper for this at the fileset layer that operates on the indexes?